pkg/controllers/configobservation/console/observe_consoleurl.go

```diff
@@ -20,8 +23,23 @@ func ObserveConsoleURL(genericlisters configobserver.Listers, recorder events.Re
 	listers := genericlisters.(configobservation.Listers)
 	errs := []error{}
 
+	clusterVersionConfig, err := listers.ClusterVersionLister.Get("cluster")
+	if err != nil {
+		return existingConfig, append(errs, err)
+	}
+
+	isConsoleCapabilityEnabled := false
+	for _, capability := range clusterVersionConfig.Status.Capabilities.EnabledCapabilities {
+		if capability == configv1.ClusterVersionCapabilityConsole {
+			isConsoleCapabilityEnabled = true
+		}
+	}
+
 	consoleConfig, err := listers.ConsoleLister.Get("cluster")
 	if err != nil {
+		if errors.IsNotFound(err) && isConsoleCapabilityEnabled {
+			return nil, nil
```

Comment:
Is it worth `return existingConfig, nil` here?  I'm not sure how much the caller assumes about `existingConfig` completion, but the actually-failing `return`s in surrounding lines include `existingConfig`, and there may be useful stuff in there, even in the "cluster is no-console" case.